src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,958 @@
+			else
+			{
+				index = wrapIndexIfNeeded(index);
+			}
+
+			// We need to examine all items with indices [0, m_iFocusedIndex) or (m_iFocusedIndex, nCount-1] for Down/Up keys.
+			// While index is within the range, we keep going through the item list until we are successfully able to focus an item,
+			// at which point we update the m_iFocusedIndex and break out of the loop.
+			while (0 <= index && index < (int)(nCount))
+			{
+				var spItemAsDependencyObject = Items[index] as DependencyObject;
+
+				var isFocusable = false;
+				// We determine whether the item is a focusable MenuFlyoutItem or MenuFlyoutSubItem because we want to exclude MenuSeparators here.
+				var spItem = spItemAsDependencyObject as MenuFlyoutItem;
+				if (spItem != null)
+				{
+					isFocusable = spItem.IsFocusable;
+				}
+				else
+				{
+					MenuFlyoutSubItem spSubItem;
+					spSubItem = spItemAsDependencyObject as MenuFlyoutSubItem;
+					if (spSubItem != null)
+					{
+						isFocusable = spSubItem.IsFocusable;
+					}
+				}
+
+				// If the item is focusable, move the focus to it, update the m_iFocusedIndex and break out of the loop.
+				if (isFocusable)
+				{
+					var spSubItem = spItemAsDependencyObject as Control;
+					if (spSubItem != null)
+					{
+						var wasFocused = spSubItem.Focus(focusState);
+						m_iFocusedIndex = index;
+						break;
+					}
+				}
+
+				// If we've gone all the way around the list of items and still have not found a suitable focus candidate,
+				// then we'll stop - there's nothing else for us to do.
+				if (index == originalFocusedIndex)
+				{
+					break;
+				}
+
+				index += deltaIndex;
+
+				// If we should be wrapping, then we'll perform the wrap at this point.
+				index = wrapIndexIfNeeded(index);
+			}
+
+		}
+
+		internal void HandleKeyDownLeftOrEscape()
+		{
+			(this as IMenuPresenter).CloseSubMenu();
+		}
+
+
+		protected override void PrepareContainerForItemOverride(
+			 DependencyObject pElement,
+			 object pItem)
+		{
+			base.PrepareContainerForItemOverride(pElement, pItem);
+
+			var spMenuFlyoutItemBase = pElement as MenuFlyoutItemBase;
+
+			spMenuFlyoutItemBase.SetParentMenuFlyoutPresenter(this);
+		}
+
+
+		protected override void ClearContainerForItemOverride(
+			 DependencyObject pElement,
+			 object pItem)
+		{
+			base.ClearContainerForItemOverride(pElement, pItem);
+
+			(pElement as MenuFlyoutItemBase).SetParentMenuFlyoutPresenter(null);
+		}
+
+		// Get the parent MenuFlyout.
+		internal MenuFlyout GetParentMenuFlyout()
+		{
+			return m_wrParentMenuFlyout?.Target as MenuFlyout;
+		}
+
+		// Sets the parent MenuFlyout.
+		internal void SetParentMenuFlyout(MenuFlyout pParentMenuFlyout)
+		{
+			m_wrParentMenuFlyout = new WeakReference(pParentMenuFlyout);
+		}
+
+		// Called when the ItemsSource property changes.
+		protected override void OnItemsSourceChanged(DependencyPropertyChangedEventArgs args)
+		{
+			base.OnItemsSourceChanged(args);
+			var pNewValue = args.NewValue;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'args' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)